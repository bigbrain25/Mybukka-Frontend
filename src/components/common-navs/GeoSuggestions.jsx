import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import MapMarker from '../icons/MapMarker';

const GeoSuggestions = ({ suggestions, handleClick }) => (
  <Fragment>
    {suggestions.map(suggestion => (
      <div
        className="suggestion-geo-group input-group"
        onClick={handleClick}
        tabIndex={0}
        role="link"
        key={suggestion.key}
      >
        <div className="input-group-prepend">
          <span className="input-group-text location-arrow">
            <MapMarker />
          </span>
        </div>
        <h4 className="suggestion suggestion-geo text-center d-block">
          {suggestion.location}
        </h4>
      </div>
    ))}
  </Fragment>
);

export default GeoSuggestions;

GeoSuggestions.propTypes = {
  suggestions: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string))
    .isRequired,
  handleClick: PropTypes.func.isRequired
};
