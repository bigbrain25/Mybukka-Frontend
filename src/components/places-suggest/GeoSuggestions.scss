@import "../../stylebase/variables.scss";

.Primary-Color {
  color: $color-primary;
}

.suggestion-geo {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-top: 0;
  span {
    font-size: 16px;
    letter-spacing: -0.3px;
    line-height: 1.5;
  }
  .secondary-address-suggest {
    color: $color-lightslategray;
  }
}

.Suggestion-Address {
  align-items: center;
  border-bottom: 1px solid #E9E9E9;
  height: auto;
  align-self: flex-start;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  padding: 10px 0;
  background-color: $color-white;
  display: flex;
  cursor: pointer;
  justify-content: space-between;
  .suggestion-geo {
    margin: 0;
  }
  &:hover {
    background-color: $color-white;
  }
  h4 {
    display: flex;
    align-items: center;
    margin-top: 0;
    margin-bottom: 0;
  }
}

.suggestion-group-style {
  align-items: flex-start;
  border-bottom: 1px solid $color-gray;
  border-left: 1px solid $color-gray;
  border-right: 1px solid $color-gray;
  height: auto;
  align-self: flex-start;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  padding: 10px 0;
  background-color: $color-light-gray;
  display: flex;
  cursor: pointer;
  &:hover {
    background-color: #fafafa;
  }
  h4 {
    display: flex;
    align-items: center;
    margin-top: 0;
    margin-bottom: 0;
  }
}
