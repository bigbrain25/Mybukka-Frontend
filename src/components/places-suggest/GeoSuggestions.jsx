/* eslint-disable camelcase */
import React, { Fragment } from 'react';

import PropTypes from 'prop-types';

import MapMarker from '../icons/MapMarker';
import './GeoSuggestions.scss';

const GeoSuggestions = ({ handleClick, predictions }) => (
  <Fragment>
    {predictions.map((suggestion) => {
      const { terms } = suggestion;
      const filteredTerms = terms.reduce((arr, item) => [...arr, item.value], []);
      return (
        <div
          className="suggestion-group-style"
          onClick={() => handleClick(suggestion, true)}
          tabIndex="0"
          role="link"
          key={suggestion.id}
        >
          <div className="input-group-prepend">
            <span className="input-group-text location-arrow">
              <MapMarker />
            </span>
          </div>
          <div className="suggestion suggestion-geo text-center d-flex">
            {filteredTerms.slice(0, 2).map((term, indx) => (
              <span key={`term-${term}`} className={`${indx !== 0 ? 'secondary-address-suggest' : ''}`}>{term}</span>))}
            {filteredTerms.length > 2 && <span className="secondary-address-suggest">
              {filteredTerms.slice(2).join(', ')}
            </span>}
          </div>
        </div>
      );
    })}
  </Fragment>
);

export default GeoSuggestions;

GeoSuggestions.propTypes = {
  handleClick: PropTypes.func.isRequired,
};
