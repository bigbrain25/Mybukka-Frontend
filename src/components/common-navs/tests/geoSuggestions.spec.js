import React from 'react';
import GeoSuggestions from '../GeoSuggestions';

describe('Geo Suggestions component', () => {
  const props = {
    suggestions: [{ location: 'Mende', key: '1' }],
    handleClick: jest.fn(),
  };
  const wrapper = shallow(<GeoSuggestions {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it(`calls the handleClick function when the
  clickable element is clicked`, () => {
    wrapper.find('.suggestion-geo-group').simulate('click');

    expect(props.handleClick).toBeCalled();
  });
});
