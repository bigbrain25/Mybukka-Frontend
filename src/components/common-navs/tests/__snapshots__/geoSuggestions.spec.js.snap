// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Geo Suggestions component renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GeoSuggestions
    handleClick={[MockFunction]}
    suggestions={
      Array [
        Object {
          "key": "1",
          "location": "Mende",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="suggestion-geo-group input-group"
          onClick={[MockFunction]}
          role="link"
          tabIndex={0}
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text location-arrow"
            >
              <MapMarker />
            </span>
          </div>
          <h4
            className="suggestion suggestion-geo text-center d-block"
          >
            Mende
          </h4>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="input-group-prepend"
            >
              <span
                className="input-group-text location-arrow"
              >
                <MapMarker />
              </span>
            </div>,
            <h4
              className="suggestion suggestion-geo text-center d-block"
            >
              Mende
            </h4>,
          ],
          "className": "suggestion-geo-group input-group",
          "onClick": [MockFunction],
          "role": "link",
          "tabIndex": 0,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="input-group-text location-arrow"
              >
                <MapMarker />
              </span>,
              "className": "input-group-prepend",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MapMarker />,
                "className": "input-group-text location-arrow",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Mende",
              "className": "suggestion suggestion-geo text-center d-block",
            },
            "ref": null,
            "rendered": "Mende",
            "type": "h4",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="suggestion-geo-group input-group"
            onClick={[MockFunction]}
            role="link"
            tabIndex={0}
          >
            <div
              className="input-group-prepend"
            >
              <span
                className="input-group-text location-arrow"
              >
                <MapMarker />
              </span>
            </div>
            <h4
              className="suggestion suggestion-geo text-center d-block"
            >
              Mende
            </h4>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text location-arrow"
                >
                  <MapMarker />
                </span>
              </div>,
              <h4
                className="suggestion suggestion-geo text-center d-block"
              >
                Mende
              </h4>,
            ],
            "className": "suggestion-geo-group input-group",
            "onClick": [MockFunction],
            "role": "link",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="input-group-text location-arrow"
                >
                  <MapMarker />
                </span>,
                "className": "input-group-prepend",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <MapMarker />,
                  "className": "input-group-text location-arrow",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Mende",
                "className": "suggestion suggestion-geo text-center d-block",
              },
              "ref": null,
              "rendered": "Mende",
              "type": "h4",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
